clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/ReadGroupWebServiceSpec.scala

```diff
@@ -35,6 +37,60 @@ class ReadGroupWebServiceSpec
     }
   }
 
+  it should "query with a project and sample and return multiple records" in {
+    val memorySearchDAO = new MemoryReadGroupSearchDAO()
+    val app = MockClioApp(searchDAO = memorySearchDAO)
+    val webService = new MockReadGroupWebService(app)
+    Post(
+      "/metadata/v1/barcodeGCP/5/LibraryGCP/GCP",
+      Map("project" -> "testProject1", "sample_alias" -> "sample1")
+    ) ~> webService.postMetadata ~> check {
+      status shouldEqual StatusCodes.OK
+    }
+    Post(
+      "/metadata/v1/barcodeGCP/6/LibraryGCP/GCP",
+      Map("project" -> "testProject1", "sample_alias" -> "sample1")
+    ) ~> webService.postMetadata ~> check {
+      status shouldEqual StatusCodes.OK
+    }
+    Post(
+      "/metadata/v1/barcodeGCP/7/LibraryGCP/GCP",
+      Map("project" -> "testProject1", "sample_alias" -> "sample2")
+    ) ~> webService.postMetadata ~> check {
+      status shouldEqual StatusCodes.OK
+    }
+
+    Post(
+      "/query/v1",
+      Map("project" -> "testProject1")
+    ) ~> webService.query ~> check {
+      memorySearchDAO.queryReadGroupCalls should have length 1
```

Comment:
Wot. If the mock DAO doesn't currently return what it had recorded when queried, I think it should be extended to do so. Otherwise there's a whole code path for result serialization in the list endpoint that we aren't unit-testing.